test/integration/graph/core/test_auto_package_type.py

```diff
@@ -43,3 +45,19 @@ def test_auto_package_type(conanfile):
     assert "package_type: header-library" in c.out
     c.run("graph info . --filter package_type -o header_only=True -o shared=False")
     assert "package_type: header-library" in c.out
+
+def test_package_type_and_header_library():
+    """ Show that forcing a package_type and header_only=True does not change the package_type"""
```

Comment:
Test is correct, the ``package_type`` should have priority.
I wonder if a warning for this is possible and not difficult, it might be worth it.